code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -267,7 +267,9 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 		assert_gas(gas_id)
 		other.assert_gas(gas_id)
 		//math is under the assumption temperatures are equal
-		if(abs(gases[gas_id][MOLES] / volume - other.gases[gas_id][MOLES] / other.volume) > min_p_delta / (R_IDEAL_GAS_EQUATION * temperature))
+		// we check for either volume being zero, as this will cause division by zero runtimes.
+		// they can have zero volume due to one not having the gas, and then we assert it initializing it with zero moles.
+		if(volume == 0 || other.volume == 0 || abs(gases[gas_id][MOLES] / volume - other.gases[gas_id][MOLES] / other.volume) > min_p_delta / (R_IDEAL_GAS_EQUATION * temperature))
```

Comment:
Cache volume and other.volume in a var instead of de-referencing them twice

Reply:
I think the volume checks could just be an early return too. Don't need to check for the volume for every single gas when it will not change.